### ISSUE TITLE
performance.now in worker

### ISSUE THREAD

**Issue by dakom** [rustwasm/wasm-bindgen#1752]
MDN says that WorkerGlobalScope has `performance` in Chrome and Firefox: https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/performance

`web-sys` doesn't seem to export this: https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.WorkerGlobalScope.html

Is this expected? If so, what is the right way to get `performance.now` from inside a worker?

**Comment by fitzgen** [comment]
It looks like our webidl, which is derived from Firefox/Gecko, only has it on the `Window` interface, not on `WorkerGlobalScope`. I wonder if this is supported in workers cross browser?

**Comment by jrandall** [comment]
It looks like performance on worker global scope is part of the High Resolution Time Level 2 recommendation, and it is defined as an attribute on the `WindowOrWorkerGlobalScope` mixin rather than directly on `WorkerGlobalScope` itself: https://www.w3.org/TR/hr-time-2/#the-performance-attribute 

The link from that spec to caniuse.com seems to indicate most browsers have full support for it: https://caniuse.com/#feat=high-resolution-time

**Comment by knpwrs** [comment]
This issue causes the `wasm-timer` crate to break for web workers: https://github.com/tomaka/wasm-timer/issues/12

**Comment by tiby312** [comment]
As a workaround, I added this and it let me call performance.now() inside of webworker:
```
#[wasm_bindgen]
extern "C" {
    #[no_mangle]
    #[used]
    static performance:web_sys::Performance;
}
```

**Comment by UsernameN0tAvailable** [comment]
From which context is perfomance called from on Firefox/Geeko? I suppose it's still WorkerGlobalScope right? so it's should be an easy fix!

**Comment by allsey87** [comment]
Another work around until this is fixed could be:
```rust
let performance = js_sys::Reflect::get(&js_sys::global(), &"performance".into())
    .expect("failed to get performance from global object")
    .unchecked_into::<web_sys::Performance>();
let time = performance.time_origin() + performance.now();
```

**Comment by daxpedda** [comment]
Fixed by #3506.
